File: apriori_algorithm/apriori.R
install.packages("arules")
library(arules)

# Create dataset
transactions <- list(
  c("milk","bread","butter"),
  c("bread","butter"),
  c("milk","bread"),
  c("milk","butter")
)

# Convert to transaction format
trans <- as(transactions, "transactions")

# Apply apriori
rules <- apriori(trans, parameter=list(supp=0.5, conf=0.8))
inspect(rules)

File: apriori_algorithm/apriori.py
from mlxtend.frequent_patterns import apriori, association_rules
from mlxtend.preprocessing import TransactionEncoder
import pandas as pd

# Example transactions
dataset = [
    ['milk', 'bread', 'butter'],
    ['bread', 'butter'],
    ['milk', 'bread'],
    ['milk', 'butter']
]

# Convert to DataFrame (one-hot encoding)
te = TransactionEncoder()
te_ary = te.fit(dataset).transform(dataset)
df = pd.DataFrame(te_ary, columns=te.columns_)

# Apply Apriori
frequent_itemsets = apriori(df, min_support=0.5, use_colnames=True)
rules = association_rules(frequent_itemsets, metric="lift", min_threshold=1.0)

print(rules)